libr/main/rabin2.c

```diff
@@ -471,6 +471,48 @@ static int rabin_show_srcline(RBin *bin, ut64 at) {
 	return false;
 }
 
+/* Map Sections to Segments https://github.com/radareorg/radare2/issues/14647 */
+static int rabin_map_sections_to_segments(RBin *bin){
+	RListIter *iter;
+	RBinSection *section = NULL, *segment = NULL;
+	RList *sections = r_list_new();
+	RList *segments = r_list_new();
+	RList *tmp = r_bin_get_sections (bin);
+	printf ("Section to Segment mapping:\n");
+	printf ("Segment\tSections...\n");	
+
+	r_list_foreach(tmp, iter, section) {
+		if (section->is_segment) {
```

Comment:
ideally we should have separae callbacks to collect sections and segments

Reply:
True, I was also quite confused when I figured out r_bin_get_sections() return both sections and segments. But to change that to two separate calls, it requires lots of work and changes a lot of code that uses is_segment...and I think it worth opening a new issue about this

https://github.com/radareorg/radare2/blob/80c74d7aa72e3bb4561e0ecba1f1e6b28599db54/libr/bin/p/bin_symbols.c#L68-L84